RelationalAI/Client.cs

```diff
@@ -40,11 +40,11 @@ public class Client
         private readonly Context _context;
         private readonly ILogger _logger;
 
-        public Client(Context context, ILogger logger = null)
+        public Client(Context context, ILogger logger = null, IAccessTokenHandler accessTokenHandler = null)
         {
             _context = context;
             _logger = logger ?? new LoggerFactory().CreateLogger("RAI-SDK");
-            _rest = new Rest(context, _logger);
+            _rest = new Rest(context, _logger, accessTokenHandler);
```

Comment:
please implement the default access token handler creation here in the client, instead of in the REST layer, to keep the REST layer as thin as possible (think, as close to Curl as possible)